client/src/controllers/BulkController.ts

```diff
@@ -0,0 +1,69 @@
+import { Controller } from '@hotwired/stimulus';
+
+/**
+ * Controller that adds the ability to toggle some or all the elements, ignoring the disabled ones
+ * Additionaly, it checks if all non-disabled items are selected or not. If they are, then select all
+ *
+ * @example
+ * <div data-controller="w-bulk">
+ *   <input type="checkbox" data-w-bulk-target="all">
+ *
+ *   <div>
+ *     <input type="checkbox" data-w-bulk-target="item" disabled>
+ *     <input type="checkbox" data-w-bulk-target="item">
+ *     <input type="checkbox" data-w-bulk-target="item">
+ *   </div>
+ *
+ *   <button data-action="w-bulk#toggleAll" data-w-bulk-force-param="false">Clear all</button>
+ *   <button data-action="w-bulk#toggleAll" data-w-bulk-force-param="true">Select all</button>
+ * </div>
+ */
+export class BulkController extends Controller {
+  /** The list of targets for this controller. */
+  static targets = ['all', 'item'];
+
+  /** The list of all select-all checkbox targets for this controller. */
+  declare readonly allTargets: HTMLInputElement[];
+
+  /** The list of all item checkbox targets for this controller. */
+  declare readonly itemTargets: HTMLInputElement[];
+
+  get activeItems() {
+    return this.itemTargets.filter(({ disabled }) => !disabled);
+  }
+
+  /**
+   * Toggles the select-all checkbox based on item checkboxes.
+   */
+  toggle(): void {
+    const isAllChecked = !this.activeItems.some((item) => !item.checked);
+
+    // Update all select all checkboxes (except disabled ones)
+    this.allTargets.forEach((target) => {
+      if (!target.disabled) {
+        // eslint-disable-next-line no-param-reassign
+        target.checked = isAllChecked;
+      }
+    });
+  }
+
+  /**
+   * Toggles all item checkboxes based on select-all checkbox.
+   */
+  toggleAll(event): void {
+    const force = event.params.get('force');
+    const isChecked = force != null ? force : event.target.checked;
+
+    this.activeItems.forEach((target) => {
+      // eslint-disable-next-line no-param-reassign
+      target.checked = isChecked;
+    });
+
+    this.allTargets.forEach((target) => {
+      if (!target.disabled) {
```

Comment:
as above, no need to do this logic, let's set the select all checkboxes even if disabled.